src/main/webapp/resources/js/contexts/AdminStatisticsContext.js

```diff
@@ -0,0 +1,196 @@
+
+  // Get updated analyses usage stats for the selected time period
+  function updateAnalysesStatsTimePeriod(timePeriod) {
+    getUpdatedAdminAnalysesStatistics(timePeriod).then(res => {
+      console.log(res);
+    }).catch((message) => {
+      notification.error({ message });
+    });
+  }
+
+  // Get updated sample usage stats for the selected time period
+  function updateSampleStatsTimePeriod(timePeriod) {
+    getUpdatedAdminSampleStatistics(timePeriod).then(res => {
+      console.log(res);
+    }).catch((message) => {
+      notification.error({ message });
+    });
+  }
+
+  /*
+   * Gets the config required for the chart
+   * @param chartType - The type of chart (bar, column, line, or pie)
+   * @param data - The data for the chart
+   * @param statsType - The type of statistics (projects, analyses, samples, users)
+   * @param timePeriod - The time period for the statistics
+   */
+  function getChartConfig(chartType, statsType, timePeriod)
+  {
+    let data = null;
+    const timePeriodText = timePeriodMap[timePeriod];
+    const isBarChartType = chartType === chartTypes.BAR;
+    const isPieDonutChartType = chartType === chartTypes.DONUT || chartType === chartTypes.PIE;
+
+    let chartTitle = "";
+    let chartAxisAlias = "";
+
+    if(statsType === statisticTypes.ANALYSES) {
+      chartTitle = `Number of analyses run in past ${timePeriodText}`;
+      chartAxisAlias = '# of Analyses';
+    } else if (statsType === statisticTypes.PROJECTS) {
+      chartTitle = `Number of projects created in past ${timePeriodText}`;
+      chartAxisAlias ='# of Projects';
+    } else if (statsType === statisticTypes.SAMPLES)
+    {
+      chartTitle = `Number of samples created in past ${timePeriodText}`;
+      chartAxisAlias = '# of Samples';
+    } else if (statsType === statisticTypes.USERS) {
+      chartTitle =`Number of users created in past ${timePeriodText}`;
+      chartAxisAlias = '# of Users';
+    }
+
+
+    // The configuration required to display a chart
+    const chartConfig = {
+      title: { visible: true, text: chartTitle },
+      forceFit: true,
+      data: data !== null  ? data : [{key:"", value:""}],
+      padding: 'auto',
+      xField: isBarChartType ? 'value' : 'key',
+      yField: isBarChartType ? 'key' : 'value',
+      meta: { key: { alias: 'Time Period' }, value: { alias: chartAxisAlias } },
+      angleField:"value",
+      label: {
+        visible: data !== null ? (isPieDonutChartType ? false : true) : false,
+        position: isBarChartType ? 'right' : 'middle',
+        adjustColor: true,
+        style: { fill: '#0D0E68', fontSize: 12, fontWeight: 600, opacity: 0.3 },
+      },
+      colorField: "key",
+      legend: {
+        visible: data !== null ? true : false,
+        position: 'bottom-center',
+      },
+      statistic: {
+        visible: data !== null ? true : false,
+        content: {
+          value: "",
+          name: '',
+        },
+      },
+    };
+
+    return chartConfig;
```

Comment:
I really don't like so many ternary statements, we are starting to see lots of this because we have so many complex objects.  I think a better way to handle this, and to make it easier to read would be:

```javascript
import merge from "lodash/merge";

function getChartConfig(chartType, statsType, timePeriod) {
    let data = null;
    const timePeriodText = timePeriodMap[timePeriod];

    let title = "";
    let chartAxisAlias = "";

    const custom = {
      [chartTypes.BAR]: {
        xField: "value",
        yField: "key",
        label: {
          position: "right",
        },
      },
      [chartTypes.DONUT]: {
        visible: false,
      },
      [chartTypes.PIE]: {
        visible: false,
      },
      [chartTypes.COLUMN]: {},
      [chartTypes.LINE]: {},
    };

    if (statsType === statisticTypes.ANALYSES) {
      title = `Number of analyses run in past ${timePeriodText}`;
      chartAxisAlias = "# of Analyses";
    } else if (statsType === statisticTypes.PROJECTS) {
      title = `Number of projects created in past ${timePeriodText}`;
      chartAxisAlias = "# of Projects";
    } else if (statsType === statisticTypes.SAMPLES) {
      title = `Number of samples created in past ${timePeriodText}`;
      chartAxisAlias = "# of Samples";
    } else if (statsType === statisticTypes.USERS) {
      title = `Number of users created in past ${timePeriodText}`;
      chartAxisAlias = "# of Users";
    }

    const config = {
      title: { visible: true, title },
      forceFit: true,
      data: data || [{ key: "", value: "" }],
      padding: "auto",
      xField: "key",
      yField: "value",
      meta: { key: { alias: "Time Period" }, value: { alias: chartAxisAlias } },
      angleField: "value",
      label: {
        visible: Boolean(data),
        position: "middle",
        adjustColor: true,
        style: { fill: "#0D0E68", fontSize: 12, fontWeight: 600, opacity: 0.3 },
      },
      colorField: "key",
      legend: {
        visible: Boolean(data),
        position: "bottom-center",
      },
      statistic: {
        visible: Boolean(data),
        content: {
          value: "",
          name: "",
        },
      },
    };

    return merge(config, [custom[chartType]]);
}
```

Reply:
I also do not think this needs to be in the context.  It could be in a separate file and just called.  I don't think there is anything in here that is directly needed from the context.

Reply:
Updated and moved into a new file in ff7ec157807d1b1bc99e6e65236ed77a1bcee291